src/orion/algo/robo/ablr/ablr_model.py

```diff
@@ -0,0 +1,589 @@
+            self.y_mean = y_train.mean(0)
+        if self.y_var == 1:
+            y_var = y_train.var(0)
+            self.y_var = torch.maximum(y_var, torch.zeros_like(y_var) + 1e-5)
+
+        assert x_train.shape[-1] == self.x_mean.shape[-1], (x_train.shape, self.x_mean)
+        x_train = torch.as_tensor(x_train, dtype=torch.float)
+        y_train = torch.as_tensor(y_train, dtype=torch.float)
+        x_train = self._normalize_x(x_train) if self.normalize_inputs else x_train
+        y_train = self._normalize_y(y_train) if self.normalize_inputs else y_train
+        neg_mll, predictive_distribution_fn = self.predictive_mean_and_variance(
+            x_train, y_train
+        )
+
+        def predict_normalized(x_test: Tensor | np.ndarray) -> Normal:
+            x_test = torch.as_tensor(x_test, dtype=torch.float)
+            if self.normalize_inputs:
+                x_test = self._normalize_x(x_test)
+            y_pred_dist = predictive_distribution_fn(x_test)
+            y_pred_dist = self._unnormalize_y(y_pred_dist)
+            return y_pred_dist
+
+            # y_pred = y_pred_dist.rsample()
+            # return self.unnormalize_y(y_pred)
+
+        return neg_mll, predict_normalized
+
+    def _get_loss(self, x: np.ndarray, y: np.ndarray) -> Tensor:
+        """Gets the loss for the given input and target."""
+        return self(x, y)[0]
+
+    @overload
+    def train(self, mode: bool = True) -> ABLR:
+        """Set the model in train mode following (as a nn.Module subclass)."""
+        ...
+
+    # pylint: disable=arguments-differ
+    @overload
+    def train(self, X: np.ndarray, y: np.ndarray, do_optimize: bool = False) -> None:
+        """Train the algorithm (as a RoBO Model subclass)."""
+        ...
+
+    # pylint: disable=signature-differs
+    def train(
+        self,
+        *args,
+        **kwargs,
+    ):
+        if len(args) == 1 and isinstance(args[0], bool):
+            mode: bool = args[0]
+            return nn.Module.train(self, mode=mode)
+        if "mode" in kwargs:
+            mode: bool = kwargs["mode"]
+            return nn.Module.train(self, mode=mode)
+        bound_args = inspect.signature(self._train_robo).bind(*args, **kwargs)
+        bound_args.apply_defaults()
+        return self._train_robo(
+            X=bound_args.arguments["X"],
+            y=bound_args.arguments["y"],
+            do_optimize=bound_args.arguments["do_optimize"],
+        )
+
+    def _train_robo(
+        self, X: np.ndarray, y: np.ndarray, do_optimize: bool | None = None
+    ) -> None:
+        """Training method for the algorithm, as a RoBO Model subclass."""
+
+        if X is None or y is None:
+            raise RuntimeError(
+                "Invalid arguments. X and y can't be None, they should be numpy arrays."
+            )
+
+        # Save the dataset so we can use it to predict the mean and variance later.
+        # NOTE: This do_optimize isn't used, but is a parameter of the Robo base class.
```

Comment:
This comment is a bit confusing. Maybe should be part of the docstring?

Reply:
Fixed: e7d238a3cdb60fe91b6a0109b4d4627a31f2c575